src/test/java/redis/clients/jedis/JedisPooledTest.java

```diff
@@ -243,6 +243,9 @@ public void cleanUp() {
       assertEquals(1, prepareCount.get());
       assertEquals(1, cleanupCount.get());
 
+      // clear existing `noauth` links
+      pool.getPool().clear();
+
```

Comment:
We don't need this. The issue is with test itself but hidden due to `.clientName("my_shiny_client_name")`. Please check #3357 
```suggestion
```